utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,492 @@
+					let extrinsic = UncheckedExtrinsic::new_signed(call, address, signature.into(), extra);
+					log::debug!(
+						target: crate::LOG_TARGET, "constructed extrinsic {}",
+						sp_core::hexdisplay::HexDisplay::from(&extrinsic.encode())
+					);
+					extrinsic
+				}
+			}
+		}}
+	};
+}
+
+// NOTE: we might be able to use some code from the bridges repo here.
+fn signed_ext_builder_polkadot(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> polkadot_runtime_exports::SignedExtra {
+	use polkadot_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+		runtime_common::claims::PrevalidateAttests::<Runtime>::new(),
+	)
+}
+
+fn signed_ext_builder_kusama(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> kusama_runtime_exports::SignedExtra {
+	use kusama_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+	)
+}
+
+fn signed_ext_builder_westend(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> westend_runtime_exports::SignedExtra {
+	use westend_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+	)
+}
+
+construct_runtime_prelude!(polkadot);
+construct_runtime_prelude!(kusama);
+construct_runtime_prelude!(westend);
+
+// NOTE: this is no longer used extensively, most of the per-runtime stuff us delegated to
+// `construct_runtime_prelude` and macro's the import directly from it. A part of the code is also
+// still generic over `T`. My hope is to still make everything generic over a `Runtime`, but sadly
+// that is not currently possible as each runtime has its unique `Call`, and all Calls are not
+// sharing any generic trait. In other words, to create the `UncheckedExtrinsic` of each chain, you
+// need the concrete `Call` of that chain as well.
+#[macro_export]
+macro_rules! any_runtime {
+	($($code:tt)*) => {
+		unsafe {
+			match $crate::RUNTIME {
```

Comment:
Not something to change for this draft, but in the future it would be nice to not have to use this `RUNTIME` and instead figure out a way to explicitly pass it in as an arg. Maybe even there is some builder model we could come up with?